src/ApiService/ApiService/onefuzzlib/JobResultOperations.cs

```diff
@@ -0,0 +1,128 @@
+                double newCovered = resultValue["covered"];
+                double newTotalCovered = resultValue["features"];
+                double newCoverageRate = resultValue["rate"];
+                newResult = result with { InstructionsCovered = newCovered, TotalInstructions = newTotalCovered, CoverageRate = newCoverageRate };
+                break;
+            case JobResultType.RuntimeStats:
+                _logTracer.LogInformation("Attempting update to iterations.");
+                double newTotalIterations = resultValue["total_count"];
+                _logTracer.LogInformation($"Attempting update to iterations {newTotalIterations}.");
+                // int newExecsSec = resultValue["execs_sec"];
+                newResult = result with { IterationCount = newTotalIterations };
+                break;
+            default:
+                _logTracer.LogInformation($"Invalid Field {type}.");
+                break;
+        }
+        _logTracer.LogInformation($"Attempting to log new result: {newResult}");
+        return newResult;
+    }
+
+    public async Async.Task<bool> TryUpdate(Job job, JobResultType resultType, Dictionary<string, double> resultValue) {
+        var jobId = job.JobId;
+        _logTracer.LogInformation($"Inside try, but before.");
+
+        var jobResult = await GetJobResult(jobId);
+
+        _logTracer.LogInformation($"Inside try.");
+        if (jobResult == null) {
+            _logTracer.LogInformation("Creating new JobResult for Job {JobId}", jobId);
+
+            var entry = new JobResult(JobId: jobId, Project: job.Config.Project, Name: job.Config.Name, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+
+            jobResult = UpdateResult(entry, resultType, resultValue);
+
+            var r = await Insert(jobResult);
+            if (!r.IsOk) {
+                _logTracer.AddHttpStatus(r.ErrorV);
+                _logTracer.LogError("failed to insert job result {JobId}", jobResult.JobId);
+                throw new InvalidOperationException($"failed to insert job result {jobResult.JobId}");
+            }
+            _logTracer.LogInformation("created job result {JobId}", jobResult.JobId);
+        } else {
+            _logTracer.LogInformation("Updating existing JobResult entry for Job {JobId}", jobId);
+
+            jobResult = UpdateResult(jobResult, resultType, resultValue);
+
+            var r = await Update(jobResult);
+            if (!r.IsOk) {
+                _logTracer.AddHttpStatus(r.ErrorV);
+                _logTracer.LogError("failed to update job result {JobId}", jobResult.JobId);
+                throw new InvalidOperationException($"failed to insert job result {jobResult.JobId}");
+            }
+            _logTracer.LogInformation("updated job result {JobId}", jobResult.JobId);
+        }
+
+        return true;
+    }
+
+    public async Async.Task<OneFuzzResult<bool>> CreateOrUpdate(Guid jobId, JobResultType resultType, Dictionary<string, double> resultValue) {
```

Comment:
I don't think it's possible for us to return `OneFuzzResult<bool>.Ok(false)` here, maybe this should return a `OneFuzzResultVoid`?